components/brave_user_agent/browser/brave_user_agent_component_installer.cc

```diff
@@ -0,0 +1,162 @@
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "crypto/sha2.h"
+
+using brave_component_updater::BraveOnDemandUpdater;
+
+namespace brave_user_agent {
+
+namespace {
+
+class BraveUserAgentComponentInstallerPolicy
+    : public component_updater::ComponentInstallerPolicy {
+ public:
+  explicit BraveUserAgentComponentInstallerPolicy(
+      const std::string& component_public_key,
+      const std::string& component_id,
+      const std::string& component_name,
+      OnComponentReadyCallback callback);
+  ~BraveUserAgentComponentInstallerPolicy() override;
+
+  BraveUserAgentComponentInstallerPolicy(
+      const BraveUserAgentComponentInstallerPolicy&) = delete;
+  BraveUserAgentComponentInstallerPolicy& operator=(
+      const BraveUserAgentComponentInstallerPolicy&) = delete;
+
+  // component_updater::ComponentInstallerPolicy
+  bool SupportsGroupPolicyEnabledComponentUpdates() const override;
+  bool RequiresNetworkEncryption() const override;
+  update_client::CrxInstaller::Result OnCustomInstall(
+      const base::Value::Dict& manifest,
+      const base::FilePath& install_dir) override;
+  void OnCustomUninstall() override;
+  bool VerifyInstallation(const base::Value::Dict& manifest,
+                          const base::FilePath& install_dir) const override;
+  void ComponentReady(const base::Version& version,
+                      const base::FilePath& path,
+                      base::Value::Dict manifest) override;
+  base::FilePath GetRelativeInstallDir() const override;
+  void GetHash(std::vector<uint8_t>* hash) const override;
+  std::string GetName() const override;
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+  bool IsBraveComponent() const override;
+
+ private:
+  const std::string component_id_;
+  const std::string component_name_;
+  OnComponentReadyCallback ready_callback_;
+  std::array<uint8_t, crypto::kSHA256Length> component_hash_;
+};
+
+BraveUserAgentComponentInstallerPolicy::BraveUserAgentComponentInstallerPolicy(
+    const std::string& component_public_key,
+    const std::string& component_id,
+    const std::string& component_name,
+    OnComponentReadyCallback callback)
+    : component_id_(component_id),
+      component_name_(component_name),
+      ready_callback_(callback) {
+  // Generate hash from public key.
+  std::string decoded_public_key;
+  base::Base64Decode(component_public_key, &decoded_public_key);
+  crypto::SHA256HashString(decoded_public_key, component_hash_.data(),
+                           component_hash_.size());
+}
+
+BraveUserAgentComponentInstallerPolicy::
+    ~BraveUserAgentComponentInstallerPolicy() = default;
+
+bool BraveUserAgentComponentInstallerPolicy::
+    SupportsGroupPolicyEnabledComponentUpdates() const {
+  return true;
+}
+
+bool BraveUserAgentComponentInstallerPolicy::RequiresNetworkEncryption() const {
+  return false;
```

Comment:
nit: We should add a comment here on why this is false. I'm not sure if it's SSL related, but maybe it should require it.